scripts/build_source_deb.py

```diff
@@ -20,33 +20,45 @@
 import os
 import subprocess
 
-def get_buildpackage_version():
-    '''
-    Return the installed version of git-buildpackage as a tuple of three ints
-    '''
-    version_output = subprocess.check_output(['git-buildpackage', '--version'])
-    m = re.match('git-buildpackage\s+([0-9]+)\.([0-9]+)\.([0-9]+)\s*', version_output)
-    if m:
-        return tuple([int(x) for x in m.groups()])
-    return None
-
-# Parse command line args
-rosdistro, package, release_version = sys.argv[1:4]
-gbp_args = sys.argv[4:]
-
-# Create a git-buildpackage command appropriate for this slave
-version = get_buildpackage_version()
-print 'Detected git-buildpackage version' + str(version)
-if version[1] >= 6:
-    # This is a new version of git-buildpackage. It knows about the --git-upstream-tree arg
-    command = ['git-buildpackage', '-S', '--git-upstream-tree=TAG',
+def _get_package_subfolders(basepath, debian_package_name):
+    subfolders = []
+    for filename in os.listdir(basepath):
+        path = os.path.join(basepath, filename)
+        if not os.path.isdir(path):
+            continue
+        if filename.startswith('%s-' % debian_package_name):
+            subfolders.append(path)
+    return subfolders
+
+rosdistro, package, release_version, workdir = sys.argv[1:5]
```

Comment:
By the way I get following error:

/home/buildbot/rosbuilder1/arbotix_indigo_trusty_amd64_debbuild/build_source_deb.py indigo arbotix_msgs 0.10.0-0 -uc -us --git-ignore-branch --git-ignore-new --git-verbose --git-dist=trusty --git-arch=amd64
 in dir /home/buildbot/rosbuilder1/arbotix_indigo_trusty_amd64_debbuild/build (timeout 1200 secs)
 watching logfiles {}
 argv: ['/home/buildbot/rosbuilder1/arbotix_indigo_trusty_amd64_debbuild/build_source_deb.py', 'indigo', 'arbotix_msgs', '0.10.0-0', '-uc', '-us', '--git-ignore-branch', '--git-ignore-new', '--git-verbose', '--git-dist=trusty', '--git-arch=amd64']
 environment:
  HOME=/home/buildbot
  HUSHLOGIN=FALSE
  LANG=en_US.UTF-8
  LESSCLOSE=/usr/bin/lesspipe %s %s
  LESSOPEN=| /usr/bin/lesspipe %s
  LOGNAME=buildbot
  LS_COLORS=rs=0:di=01;34:ln=01;36:mh=00:pi=40;33:so=01;35:do=01;35:bd=40;33;01:cd=40;33;01:or=40;31;01:mi=00:su=37;41:sg=30;43:ca=30;41:tw=30;42:ow=34;42:st=37;44:ex=01;32:*.tar=01;31:*.tgz=01;31:*.arc=01;31:*.arj=01;31:*.taz=01;31:*.lha=01;31:*.lz4=01;31:*.lzh=01;31:*.lzma=01;31:*.tlz=01;31:*.txz=01;31:*.tzo=01;31:*.t7z=01;31:*.zip=01;31:*.z=01;31:*.Z=01;31:*.dz=01;31:*.gz=01;31:*.lrz=01;31:*.lz=01;31:*.lzo=01;31:*.xz=01;31:*.bz2=01;31:*.bz=01;31:*.tbz=01;31:*.tbz2=01;31:*.tz=01;31:*.deb=01;31:*.rpm=01;31:*.jar=01;31:*.war=01;31:*.ear=01;31:*.sar=01;31:*.rar=01;31:*.alz=01;31:*.ace=01;31:*.zoo=01;31:*.cpio=01;31:*.7z=01;31:*.rz=01;31:*.cab=01;31:*.jpg=01;35:*.jpeg=01;35:*.gif=01;35:*.bmp=01;35:*.pbm=01;35:*.pgm=01;35:*.ppm=01;35:*.tga=01;35:*.xbm=01;35:*.xpm=01;35:*.tif=01;35:*.tiff=01;35:*.png=01;35:*.svg=01;35:*.svgz=01;35:*.mng=01;35:*.pcx=01;35:*.mov=01;35:*.mpg=01;35:*.mpeg=01;35:*.m2v=01;35:*.mkv=01;35:*.webm=01;35:*.ogm=01;35:*.mp4=01;35:*.m4v=01;35:*.mp4v=01;35:*.vob=01;35:*.qt=01;35:*.nuv=01;35:*.wmv=01;35:*.asf=01;35:*.rm=01;35:*.rmvb=01;35:*.flc=01;35:*.avi=01;35:*.fli=01;35:*.flv=01;35:*.gl=01;35:*.dl=01;35:*.xcf=01;35:*.xwd=01;35:*.yuv=01;35:*.cgm=01;35:*.emf=01;35:*.ogv=01;35:*.ogx=01;35:*.aac=00;36:*.au=00;36:*.flac=00;36:*.m4a=00;36:*.mid=00;36:*.midi=00;36:*.mka=00;36:*.mp3=00;36:*.mpc=00;36:*.ogg=00;36:*.ra=00;36:*.wav=00;36:*.oga=00;36:*.opus=00;36:*.spx=00;36:*.xspf=00;36:
  MAIL=/var/mail/buildbot
  PATH=/home/buildbot/buildbot-ros/scripts:/home/buildbot/buildbot-env/bin:/home/buildbot/bin:/home/buildbot/.local/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/usr/games:/usr/local/games:/snap/bin
  PS1=(buildbot-env) ${debian_chroot:+($debian_chroot)}\u@\h:\w\$ 
  PWD=/home/buildbot/rosbuilder1/arbotix_indigo_trusty_amd64_debbuild/build
  SHELL=/bin/bash
  SHLVL=1
  TERM=linux
  USER=buildbot
  VIRTUAL_ENV=/home/buildbot/buildbot-env
  XDG_RUNTIME_DIR=/run/user/1000
  XDG_SEAT=seat0
  XDG_SESSION_ID=1
  XDG_VTNR=1
  _=/home/buildbot/buildbot-env/bin/buildslave
 using PTY: False
Invoking 'gbp buildpackage --git-ignore-new --git-ignore-branch -S -us -uc --git-upstream-tree=TAG --git-upstream-tag=release/indigo/arbotix_msgs/0.10.0-0 -us --git-ignore-branch --git-ignore-new --git-verbose --git-dist=trusty --git-arch=amd64' in '-uc/build'
Traceback (most recent call last):
  File "/home/buildbot/rosbuilder1/arbotix_indigo_trusty_amd64_debbuild/build_source_deb.py", line 58, in <module>
    subprocess.check_call(cmd,cwd=sources_dir,env=env)
  File "/usr/lib/python2.7/subprocess.py", line 536, in check_call
    retcode = call(*popenargs, **kwargs)
  File "/usr/lib/python2.7/subprocess.py", line 523, in call
    return Popen(*popenargs, **kwargs).wait()
  File "/usr/lib/python2.7/subprocess.py", line 711, in __init__
    errread, errwrite)
  File "/usr/lib/python2.7/subprocess.py", line 1343, in _execute_child
    raise child_exception
OSError: [Errno 2] No such file or directory: '-uc/build'
program finished with exit code 1